src/herder/TxSetFrame.h

```diff
@@ -340,7 +340,8 @@ class TxSetPhaseFrame
     // Returns a copy of this phase with transactions sorted for apply.
     TxSetPhaseFrame sortedForApply(Hash const& txSetHash) const;
     bool checkValid(Application& app, uint64_t lowerBoundCloseTimeOffset,
-                    uint64_t upperBoundCloseTimeOffset) const;
+                    uint64_t upperBoundCloseTimeOffset,
+                    bool isAlreadyValidated = false) const;
```

Comment:
Nit: please add a comment about isAlreadyValidated. It might be nice to rename to txsAlreadyValidated, as it currently reads like we are redundantly calling checkValid on an object that's already been validated.